Automattic/jetpack
# Featured content: fail gracefully when taxs are filtered out

#### Changes proposed in this Pull Request:

In some scenarios, plugins or themes using the get_terms filter may not return the array of taxonomies we expect in Featured Content.
In those situations, let's fail gracefully to avoid notices such as:

```
Warning: in_array() expects parameter 2 to be array, null given in wp-content/plugins/jetpack/modules/theme-tools/featured-content.php on line 394
```

This was originally reported in the forums:
https://wordpress.org/support/topic/after-updating-to-wordpress-5-7-getting-jetpack-error/

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

Not too sure how to reproduce yet, you would need your own snippet to filter `get_terms` but not take the second parameter into account properly.
To ensure Featured Content still works, try the following:

* Set up a new site with a theme like https://wordpress.org/themes/superhero/
* Publish a few posts with large featured images
* Go to Appearance > Customize and check the Featured Content settings
* Make sure they still work, that you can exclude a post if it has a specific tag).
